funcx_sdk/funcx/sdk/asynchronous/polling_task.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+from asyncio import AbstractEventLoop
+import time
+from funcx.sdk.asynchronous.funcx_task import FuncXTask
+
+
+class PollingTask:
+    """
+    Launches an asycnhio task for polling unresolved funcX tasks. Uses a work queue to
+    record unresolved funcX Tasks. Pulls one off and attempts to retrieve the result.
+    If this attempt fails, put the value back on the queue for later attempt.
+    Use a throttle to insure we DDOS the web service
+    """
+    async def poll_funcx(self, queue: asyncio.Queue):
+        while True:
+            t = await queue.get()
+            try:
+                result = self.fxc.get_result(t.task_id)
```

Comment:
This already looks good. We could eke out more performance using batch queries.

Reply:
If FuncX supports batch queries - then that makes a huge difference - just ask for all the results at once, and pass on those that are ready.